lib/util/cache/repository/types.ts

```diff
@@ -52,4 +52,5 @@ export interface Cache {
     };
   };
   gitConflicts?: GitConflictsCache;
+  renovateRefsFetchSkipsCounter?: number;
```

Comment:
I would prefer to avoid "churn" in the cache unnecessarily because it will trigger it to be pushed/analyzed by the app backend any time it changes. Can we remove this 15 runs logic? I don't think there's much at risk?

Reply:
Not so much risk, but if ref wasn't removed with finalize, it will never be removed in future. How about random cleanup with `P=0.05`?

Reply:
Sure, will be the simplest solution. Let's return to the optimization once it's the actual problem.